components/cell/io.cellery.security.cell.sts.server/src/main/java/io/cellery/security/cell/sts/server/core/CellStsUtils.java

```diff
@@ -101,7 +101,11 @@ public static String getPrettyPrintJson(Map<String, String> attributes) {
      */
     public static String getIssuerName(String cellName) {
 
-        return cellName + "--sts-service";
+        if (cellName.equals(Constants.COMPOSITE_CELL_NAME)) {
+            return cellName + Constants.STS_SERVICE + "." + Constants.SYSTEM_NAMESPACE;
```

Comment:
Can we use StringBuilder for these concatenation ? It will be better.

Reply:
Done